### ISSUE TITLE
Skaffold should provide more detail on parsing errors

### ISSUE THREAD

**Issue by briandealwis** [GoogleContainerTools/skaffold#5659]
Invoking Skaffold with a `skaffold.yaml` produces a terse error message with no detail as the problem.
```
parsing skaffold config: error parsing skaffold configuration file
```

But `skaffold fix` will pinpoint the problematic line:
```
unable to parse config: yaml: unmarshal errors:
  line 4: field pimage not found in type latest.Artifact
```

We should report the problematic line.

**Comment by kschaab** [comment]
I think the priority is incorrect on this as this error is also returned from `skaffold diagnose` as well.  Considering this:
```
  # Search for configuration issues and print the effective configuration
  skaffold diagnose
```
it seems like configuration issues like these should be more detailed than:
```
skaffold diagnose --yaml-only --filename skaffold.yaml
WARN[0000] Your Skaffold version might be too old. Download the latest version (1.24.0) from:
  https://github.com/GoogleContainerTools/skaffold/releases/tag/v1.24.0
parsing skaffold config: error parsing skaffold configuration file
```

**Comment by briandealwis** [comment]
Oops, This was fixed in #5722:
```
$ skaffold diagnose --yaml-only
WARN[0000] Your Skaffold version might be too old. Download the latest version (1.24.0) from:
  https://github.com/GoogleContainerTools/skaffold/releases/tag/v1.24.0 
parsing skaffold config: error parsing skaffold configuration file: parsing api version: yaml: line 4: could not find expected ':'
```
